backend_challenge.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+
+# Discount Voucher
+
+from datetime import datetime
+
+voucher={'code':"20FORME",'discount':0.2,'min_purchase':100,'expired_d':12,'expired_m':7,'expired_y':2019}
+subtotal=200.0
+shipping=0.0
+date_now=datetime.date(datetime.now())
+date=date_now.strftime("%d/%m/%Y")
+
+print("Subtotal = RM", subtotal)
+print("Shipping Fee = RM", shipping)
+print("Voucher Code: ")
+code=input()
+
+def voucher20(code, voucher, subtotal,shipping, date):
+    dd,mm,yy=date.split('/')
+    dd=int(dd)
+    mm=int(mm)
+    yy=int(yy)
+    total_amount=0
+    t1=subtotal+shipping
+    if (code == voucher['code']):
+        if (dd != 31 and mm+1 == voucher['expired_m'] and yy == voucher['expired_y']):
+            if (subtotal >= voucher['min_purchase']):
+                total_amount=t1-(t1*voucher['discount'])
+                return total_amount
+            else:
+                return t1
+        else:
+            return t1
+    else:
+        return t1
```

Comment:
That's a repetitive returns!

Reply:
`if (code == voucher['code'] and dd != 31 and mm+1 == voucher['expired_m'] and yy == voucher['expired_y'] and subtotal >= voucher['min_purchase']):
        total_amount=t1-(t1*voucher['discount'])
        return total_amount
    else:
        return t1`